### PR TITLE
Support f9-f11 as physical keys mirroring the right and left hildon buttons

### PR DESCRIPTION
**Pull request by IMbackK**
the Menu key dosent work jet, everything else (back, tasknav/launcher button) works as expected

### CODE REVIEW

**Comment by IMbackK** [comment]
together with https://github.com/IMbackK/droid4-linux/tree/touchscreen-buttons-58-clean this works great

**Comment by IMbackK** [comment]
works great with https://github.com/IMbackK/droid4-linux/tree/touchscreen-buttons-58-clean

**Comment by IMbackK** [comment]
menu key now also works

**Comment by IMbackK** [comment]
this pr now also allows you raise HIM with any window if https://github.com/maemo-leste/hildon-input-method/pull/5 is also applied. still works fine alone

**Comment by IMbackK** [comment]
This pr now also makes all shortcuts configurable via a new file /usr/share/hilodn-desktop/shortcuts.ini and generally greatly improves the shortcut code

File: INSTALL [hunk 1]
```diff
@@ -1,236 +1,368 @@
 Installation Instructions
```
Comment (freemangordon) [inline INSTALL]: Please, move INSTALL file changes to a separate commit (or remove them)

File: src/home/hd-home.c [hunk 2]
```diff
@@ -1062,13 +1062,6 @@ hd_home_desktop_key_release (XKeyEvent *xev, void *userdata)
       (priv->key_sent == KEY_SENT_NONE) && (!STATE_IS_TASK_NAV(hd_render_manager_get_state())))
       return;
 
-  if((XkbKeycodeToKeysym(clutter_x11_get_default_display(), xev->keycode, 0, 0) == GDK_Control_L) &&
```
Comment (freemangordon) [inline src/home/hd-home.c:1065]: Where is this block in hd-shortcuts.c? What it was replaced with?

File: src/main.c [hunk 3]
```diff
@@ -433,13 +227,6 @@ enumerate_input_devices (Display *dpy)
 
                   xi_dev->is_ts = (vi->mode & DeviceMode) == Absolute;
                   xi_dev->dev = dev;
-#if 0
```
Comment (freemangordon) [inline src/main.c:436]: Why is this removed?

File: src/util/hd-shortcuts.c [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+#include <matchbox/comp-mgr/mb-wm-comp-mgr.h>
+
+#include "hildon-desktop.h"
+#include "hd-wm.h"
+#include "hd-theme.h"
+#include "hd-util.h"
+#include "hd-dbus.h"
+#include "hd-volume-profile.h"
+#include "../launcher/hd-app-mgr.h"
+#include "../home/hd-render-manager.h"
```
Comment (freemangordon) [inline src/util/hd-shortcuts.c:23]: I'd rather move that file in the upper directory, that way there will be no need for "../$whatever" includes. But no strong preference.

**Review by freemangordon — CHANGES_REQUESTED** [review]
Seems there are mixed functional changes and moved code here, please either split code movement from new/changed functionality to separate commits or explain in details what has been changed (and how).

File: src/home/hd-home.c (hunk 2 above)
Comment (IMbackK) [reply src/home/hd-home.c:1065]: it was removed, action left button dose the same thing, and can be assigned to whatever key combination you want in shortcuts.ini

File: src/main.c (hunk 3 above)
Comment (IMbackK) [reply src/main.c:436]: lets not litter the code base with this kind of #if 0 disabled debug print. therfore this ill formatted debug print needs to go. if you think its needed promote it to compiled in g_debug() and improve the output formating. if its not important enough to warrant that (imo it is not) then just gain this information via gdb when needed.

File: src/util/hd-shortcuts.c (hunk 4 above)
Comment (IMbackK) [reply src/util/hd-shortcuts.c:23]: > Seems there are mixed functional changes and moved code here, please either split code movement from new/changed functionality to separate commits or explain in details what has been changed (and how).

there is some moving of code, but mostly its a refactor, im not going back and splitting this pr like that, as due to the code structure its impractical to do so while ensuring that every commit compiles for future bisecting.

**Comment by IMbackK** [comment]
This pr remvoes the hardcoded shortcuts and replaces them with actions that can be assigned any shortcut in a configuration file.
It also adds 3 actions that corrispond to the middle and and right hildon button in the title bar and a action to raise the vkb

File: INSTALL (hunk 1 above)
Comment (IMbackK) [reply INSTALL]: This is kinda a annoying one, so everytime you configure h-d this file gets updated based on the current auto tools version, so this is sort of a generated file, at the same time we want it in the repo. im not sure what to do about this.

Comment (IMbackK) [reply INSTALL]: anyhow i reverted it for now
